### ISSUE TITLE
[Question] Having trouble importing transformer_lens

### ISSUE THREAD

**Issue by arunasank** [neelnanda-io/TransformerLens#393]
### Question
```
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
Input In [123], in <cell line: 6>()
      4 from fancy_einsum import einsum
      5 from dataclasses import dataclass
----> 6 import transformer_lens

File ~/.local/lib/python3.10/site-packages/transformer_lens/__init__.py:13, in <module>
     11 from .ActivationCache import ActivationCache
     12 from .HookedTransformer import HookedTransformer
---> 13 from .SVDInterpreter import SVDInterpreter
     14 from .HookedEncoder import HookedEncoder
     15 from . import head_detector

File ~/.local/lib/python3.10/site-packages/transformer_lens/SVDInterpreter.py:5, in <module>
      3 import fancy_einsum as einsum
      4 import torch
----> 5 from typeguard import typechecked
      6 from typing_extensions import Literal
      8 from transformer_lens import FactoredMatrix, HookedTransformer

File ~/.local/lib/python3.10/site-packages/typeguard/__init__.py:4, in <module>
      1 import os
      2 from typing import Any
----> 4 from ._checkers import TypeCheckerCallable as TypeCheckerCallable
      5 from ._checkers import TypeCheckLookupCallback as TypeCheckLookupCallback
      6 from ._checkers import check_type_internal as check_type_internal

File ~/.local/lib/python3.10/site-packages/typeguard/_checkers.py:66, in <module>
     57 else:
     58     from typing_extensions import (
     59         Annotated,
     60         TypeAlias,
   (...)
     64         is_typeddict,
     65     )
---> 66     from typing_extensions import Any as SubclassableAny
     68 if sys.version_info >= (3, 10):
     69     from importlib.metadata import entry_points

ImportError: cannot import name 'Any' from 'typing_extensions' (/usr/local/lib/python3.10/dist-packages/typing_extensions.py)

```
Trouble importing the library, version 1.6.1. I'm using python 3.10.

**Comment by jbloomAus** [comment]
Can you please share your `pip freeze` output? probably a typeguard version issue.

**Comment by arunasank** [comment]
```
WARNING: The directory '/u/arunas/.cache/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you should use sudo's -H flag.
absl-py==1.2.0
accelerate==0.23.0
aesara==2.7.9
aiofiles==22.1.0
aiohttp==3.8.4
aiosignal==1.2.0
aiosqlite==0.18.0
altair==4.2.0
anyio==3.6.2
argon2-cffi==21.3.0
argon2-cffi-bindings==21.2.0
arrow==1.2.3
asttokens==2.0.5
astunparse==1.6.3
async-generator==1.10
async-timeout==4.0.2
attrs==21.2.0
Automat==20.2.0
awscli==1.22.34
Babel==2.12.1
backcall==0.2.0
bcrypt==3.2.0
beartype==0.14.1
beautifulsoup4==4.11.1
bencodepy==0.9.5
bert-serving-client==1.10.0
bert-serving-server==1.10.0
bitarray==0.8.1
bleach==5.0.1
blinker==1.4
blis==0.7.9
botocore==1.23.34
Brotli==1.0.9
bs4==0.0.1
cachetools==5.2.0
captum==0.6.0
catalogue==2.0.8
certifi==2022.9.24
cffi==1.15.1
cfscrape==2.1.1
chardet==3.0.4
charset-normalizer==2.1.0
chatgptpy==1.0.5
chromedriver-autoinstaller==0.4.0
click==8.0.3
cloudscraper==1.2.66
colorama==0.4.4
command-not-found==0.3
commonmark==0.9.1
confection==0.0.3
configobj==5.0.6
cons==0.4.5
constantly==15.1.0
cryptography==3.4.8
cssselect2==0.7.0
cycler==0.11.0
cymem==2.0.7
cypari==2.4.1
data==0.4
datasets==2.7.1
dbus-python==1.2.18
debugpy==1.6.2
decorator==5.1.1
defusedxml==0.7.1
devscripts===2.22.1ubuntu1
dill==0.3.5.1
distlib==0.3.5
distro==1.7.0
distro-info===1.1build1
docker-pycreds==0.4.0
docutils==0.17.1
einops==0.6.0
einops-exts==0.0.3
en-core-web-lg @ https://github.com/explosion/spacy-models/releases/download/en_core_web_lg-3.4.1/en_core_web_lg-3.4.1-py3-none-any.whl
entrypoints==0.4
etuples==0.3.5
evaluate==0.3.0
exceptiongroup==1.0.1
executing==0.9.1
fancy-einsum==0.0.3
fastjsonschema==2.16.1
ffmpeg-python==0.1.17
filelock==3.7.1
flamingo-mini @ file:///u/arunas/implang/transformers-from-scratch/gpt2-with-ptb/flamingo-mini
flatbuffers==23.5.8
fonttools==4.34.4
fqdn==1.5.1
frozenlist==1.3.1
fsspec==2022.10.0
funcsigs==1.0.2
future==0.18.3
fuzzywuzzy==0.18.0
FXrays==1.3.5
gast==0.4.0
geojson==2.5.0
gitdb==4.0.10
GitPython==3.1.29
glog==0.3.1
google-auth==2.18.0
google-auth-oauthlib==1.0.0
google-pasta==0.2.0
gpg===1.16.0-unknown
GPUtil==1.4.0
grpcio==1.54.0
h11==0.14.0
h2==4.1.0
h5py==2.10.0
hpack==4.0.0
html2text==2020.1.16
httpcore==0.16.3
httplib2==0.20.2
httpx==0.23.1
huggingface-hub==0.10.1
hyperframe==6.0.1
hyperlink==21.0.0
idna==3.3
imageio==2.24.0
importlib-metadata==4.6.4
incremental==21.3.0
iotop==0.6
ipykernel==6.15.1
ipython==8.4.0
ipython-genutils==0.2.0
ipywidgets==7.7.1
iso3166==2.1.1
isoduration==20.11.0
jax==0.4.9
jaxtyping==0.2.22
jedi==0.18.1
jeepney==0.7.1
Jinja2==3.1.2
jmespath==0.10.0
joblib==1.1.0
joypy==0.2.6
json5==0.9.11
jsonpointer==2.3
jsonschema==4.17.3
jupyter==1.0.0
jupyter-console==6.4.4
jupyter-contrib-core==0.4.0
jupyter-contrib-nbextensions==0.5.1
jupyter-events==0.6.3
jupyter-highlight-selected-word==0.2.0
jupyter-latex-envs==1.4.6
jupyter-nbextensions-configurator==0.5.0
jupyter-ydoc==0.2.3
jupyter_client==8.1.0
jupyter_core==5.3.0
jupyter_server==2.5.0
jupyter_server_fileid==0.8.0
jupyter_server_terminals==0.4.4
jupyter_server_ydoc==0.8.0
jupyterlab==3.6.2
jupyterlab-pygments==0.2.2
jupyterlab-widgets==1.1.1
jupyterlab_server==2.21.0
kaggle==1.5.13
kaleido==0.1.0
keras==2.12.0
Keras-Preprocessing==1.1.2
keyring==23.5.0
kiwisolver==1.4.4
kmodes==0.12.2
knot-floer-homology==1.2
langcodes==3.3.0
langdetect==1.0.9
Lasagne==0.1
latex==0.7.0
launchpadlib==1.10.16
lazr.restfulclient==0.14.4
lazr.uri==1.0.6
leveldb==0.201
libclang==14.0.6
lmdb==1.3.0
logical-unification==0.4.5
lxml==4.9.1
Markdown==3.4.1
MarkupSafe==2.1.2
matplotlib==3.5.2
matplotlib-inline==0.1.3
miniKanren==1.0.3
mistune==0.8.4
ml-dtypes==0.1.0
more-itertools==8.10.0
multidict==6.0.4
multipledispatch==0.6.0
multiprocess==0.70.13
murmurhash==1.0.9
mutagen==1.46.0
nagios-nvidia-smi-plugin==0.4.0
nagiosplugin==1.3.3
nbclassic==0.5.3
nbclient==0.6.6
nbconvert==6.5.0
nbformat==5.4.0
nest-asyncio==1.5.5
netifaces==0.11.0
networkx==2.8.5
nltk==3.8.1
notebook==6.4.12
notebook_shim==0.2.2
numexpr==2.8.4
numpy==1.23.5
nvidia-ml-py==11.515.48
oauthlib==3.2.0
olefile==0.46
onnx==1.13.0
opencv-contrib-python==4.7.0.68
opencv-python==4.7.0.68
opt-einsum==3.3.0
outcome==1.2.0
packaging==21.3
pandas==1.4.3
pandocfilters==1.5.0
parso==0.8.3
pathlib==1.0.1
pathtools==0.1.2
pathy==0.10.1
patsy==0.5.3
pexpect==4.8.0
pickleshare==0.7.5
Pillow==9.0.1
platformdirs==2.5.2
plink==2.4.1
plotly==5.13.1
plotly-express==0.4.1
ply==3.11
polyfuzz==0.4.0
polyglot==16.7.4
preshed==3.0.8
prometheus-client==0.14.1
promise==2.3
prompt-toolkit==3.0.30
protobuf==3.20.3
psutil==5.9.1
ptyprocess==0.7.0
pure-eval==0.2.2
pyarrow==9.0.0
pyasn1==0.4.8
pyasn1-modules==0.2.1
pyChatGPT==0.1.0
pycparser==2.21
pycryptodomex==3.17
pycurl==7.45.1
pydantic==1.10.2
pydeck==0.8.0
pydot==1.4.2
Pygments==2.11.2
PyGObject==3.42.1
PyHamcrest==2.0.2
PyJWT==2.3.0
Pympler==1.0.1
pyOpenSSL==21.0.0
pyparsing==2.4.7
pypng==0.20220715.0
pyrsistent==0.18.1
pyserial==3.5
PySocks==1.7.1
python-apt==2.3.0+ubuntu2.1
python-dateutil==2.8.2
python-debian===0.1.43ubuntu1
python-dotenv==0.21.0
python-gflags==3.1.2
python-json-logger==2.0.7
python-magic==0.4.24
python-slugify==8.0.1
python-snappy==0.6.1
pytz==2022.1
pytz-deprecation-shim==0.1.0.post0
PyWavelets==1.4.1
pyxdg==0.27
PyYAML==5.4.1
pyzmq==25.0.2
qtconsole==5.3.1
QtPy==2.1.0
rapidfuzz==2.13.2
regex==2022.7.25
reportlab==3.6.12
requests==2.28.2
requests-oauthlib==1.3.1
requests-toolbelt==0.10.1
responses==0.18.0
rfc3339-validator==0.1.4
rfc3986==1.5.0
rfc3986-validator==0.1.1
rich==12.6.0
roman==3.3
rsa==4.8
s3transfer==0.5.0
scikit-build==0.16.6
scikit-image==0.19.3
scikit-learn==1.1.2
scipy==1.10.1
screen-resolution-extra==0.0.0
seaborn==0.12.1
SecretStorage==3.3.1
selenium==4.6.0
semver==2.13.0
Send2Trash==1.8.0
sentence-transformers==2.2.2
sentencepiece==0.1.97
sentry-sdk==1.11.1
seqeval==1.2.2
service-identity==18.1.0
setproctitle==1.3.2
Shapely==1.8.2
shortuuid==1.0.11
shutilwhich==1.1.0
simpletransformers==0.63.9
sip==6.7.7
six==1.16.0
sklearn==0.0
smart-open==6.2.0
smmap==5.0.0
snappy==3.0.3
snappy-manifolds==1.1.2
sniffio==1.3.0
sortedcontainers==2.4.0
sos==4.4
soupsieve==2.3.2.post1
spacy==3.5.1
spacy-alignments==0.9.0
spacy-legacy==3.0.12
spacy-loggers==1.0.4
spacy-transformers==1.2.2
sparse-dot-topn==0.3.3
spherogram==2.1
srsly==2.4.5
ssh-import-id==5.11
stack-data==0.3.0
statsmodels==0.13.5
streamlit==1.15.2
svglib==1.4.1
svgling==0.3.1
svgwrite==1.4.3
systemd-python==234
tempdir==0.7.1
tenacity==8.2.1
tensorboard==2.12.3
tensorboard-data-server==0.7.0
tensorboard-plugin-wit==1.8.1
tensorflow==2.9.1
tensorflow-estimator==2.12.0
tensorflow-gpu==2.9.1
tensorflow-io-gcs-filesystem==0.26.0
termcolor==1.1.0
terminado==0.15.0
text-unidecode==1.3
tf2onnx==1.9.3
Theano==1.0.5
thinc==8.1.9
threadpoolctl==3.1.0
tifffile==2022.10.10
timm==0.6.12
tinycss2==1.1.1
tls-client==0.1.5
tokenizers==0.12.1
toml==0.10.2
tomli==2.0.1
toolz==0.12.0
torch==1.12.0+cu113
torchaudio==0.12.0+cu113
torchvision==0.13.0+cu113
tornado==6.2
tqdm==4.66.1
traitlets==5.9.0
transformer-lens==1.6.1
transformers @ git+https://github.com/huggingface/transformers@3830b3f74a57b3bcf6f14016c483fb0bb14b01ce
trio==0.22.0
trio-websocket==0.9.2
turfpy==0.0.7
Twisted==22.1.0
typeguard==4.1.5
typer==0.7.0
typing_extensions==4.8.0
tzdata==2022.7
tzlocal==4.2
ubuntu-advantage-tools==27.11.2
ubuntu-drivers-common==0.0.0
ufw==0.36.1
unattended-upgrades==0.1
unidiff==0.5.5
uri-template==1.2.0
urllib3==1.26.13
uuid==1.30
validators==0.20.0
virtualenv==20.16.2
vit-pytorch==1.0.2
wadllib==1.3.6
wandb==0.13.6
wasabi==0.10.1
watchdog==2.2.0
wcwidth==0.2.5
webcolors==1.13
webdriver-manager==3.8.4
webencodings==0.5.1
websocket-client==1.5.1
websockets==10.4
Werkzeug==2.2.1
widgetsnbextension==3.6.1
Wikipedia-API==0.5.4
wptools==0.4.17
wrapt==1.14.1
wsproto==1.2.0
xdg==5
xkit==0.0.0
xxhash==3.0.0
y-py==0.5.9
yarl==1.9.2
youtube-dl==2021.12.17
ypy-websocket==0.8.2
yt-dlp==2023.3.4
zipp==1.0.0
zope.interface==5.4.0
```

**Comment by arunasank** [comment]
Would it be possible to add a requirements.txt for TransformerLens? :)

**Comment by jbloomAus** [comment]
As far as I can tell, with a vanilla install from dev or from pip with 1.6.1, then versions of typeguard/beartype that you're using should work. Adding a requirements.txt would be redundant with the pyproject.toml which is already present. 

Can you please try to do a vanilla install in a fresh environment and see if that works? Alternatively, the docker container might be the way to go.

**Comment by alan-cooney** [comment]
Should be solved with #406 - typing_extensions was in setup.py (not meant to be used any more) but not in pyproject.toml
